datastore/longhorn.go

```diff
@@ -200,13 +200,25 @@ func (s *DataStore) applyCustomizedDefaultSettingsToDefinitions(customizedDefaul
 func (s *DataStore) syncSettingCRsWithCustomizedDefaultSettings(customizedDefaultSettings map[string]string, defaultSettingCMResourceVersion string) error {
 	for _, sName := range types.SettingNameList {
 		configMapResourceVersion := ""
-		if s, err := s.GetSettingExactRO(sName); err != nil {
+		setting, err := s.GetSettingExactRO(sName)
+		if err != nil {
 			if !apierrors.IsNotFound(err) {
 				return err
 			}
 		} else {
-			if s.Annotations != nil {
-				configMapResourceVersion = s.Annotations[types.GetLonghornLabelKey(types.ConfigMapResourceVersionKey)]
+			if setting.Annotations != nil {
+				configMapResourceVersion = setting.Annotations[types.GetLonghornLabelKey(types.ConfigMapResourceVersionKey)]
+			}
+		}
+
+		if sName == types.SettingNamePriorityClass {
+			volumesDetached, err := s.AreAllVolumesDetachedState()
+			if err != nil {
+				return errors.Wrapf(err, "failed to list volumes to check if all volumes are detached")
+			}
+			if !volumesDetached || (setting != nil && setting.Value != "") {
```

Comment:
Then, later if users update the config map directly, the setting will not be updated anymore. The source of truth should be from the config map, right?

```go
(setting != nil && setting.Value != "")
```

It seems you need to remove this condition and also still prevent this change during the upgrade.

Reply:
Sounds good, but suggest the parameter named `allowOverride bool`.

Reply:
Modified to that change the ConfigMap `longhorn-default-setting` for the setting `priority-class` if there are one or more volumes attached or the setting `priority-class` has been customized by users when doing the upgrade.

Reply:
Can't understand. Can you explain the change again?

Reply:
When the Longhorn daemon starts and walks through the upgrade path, it will check the ConfigMap CR `longhorn-default-setting`.
The setting `priority-class` in ConfigMap CR `longhorn-default-setting` will be set as the value of the setting `priority-class` in the cluster if volumes are attached or users have customized the setting `priority-class` of the cluster.
The setting 'priority-class' of the default settings in ConfigMap CR `longhorn-default-setting` will be consistent with the setting 'priority-class' of the cluster.